lib/livebook_web/live/hooks/app_auth_hook.ex

```diff
@@ -73,9 +81,20 @@ defmodule LivebookWeb.AppAuthHook do
     {:cont, socket}
   end
 
-  defp livebook_authenticated?(session, socket) do
+  defp livebook_authorized?(session, socket) do
     uri = get_connect_info(socket, :uri)
-    LivebookWeb.AuthPlug.authenticated?(session, uri.port)
+    LivebookWeb.AuthPlug.authorized?(session, uri.port)
+  end
+
+  defp app_authorized?(session, app) do
+    user =
+      LivebookWeb.UserPlug.build_current_user(
+        session,
+        session["identity_data"],
+        session["user_data"]
+      )
```

Comment:
Ah, also, let's change the hook order here:

https://github.com/livebook-dev/livebook/blob/1628ddbc51f5d4b3993a45efd7f7afc37c0c0580/lib/livebook_web/router.ex#L62

and here:

https://github.com/livebook-dev/livebook/blob/1628ddbc51f5d4b3993a45efd7f7afc37c0c0580/lib/livebook_web/router.ex#L142

The `UserHook` should go first, to match the plugs order:

https://github.com/livebook-dev/livebook/blob/1628ddbc51f5d4b3993a45efd7f7afc37c0c0580/lib/livebook_web/router.ex#L24-L28

---

Once we do that, you can use `socket.assigns.current_user` here :)